core/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -214,32 +213,49 @@ private void executeBulk(final BulkRequest bulkRequest, final long startTime, fi
         MetaData metaData = clusterState.metaData();
         for (int i = 0; i < bulkRequest.requests.size(); i++) {
             ActionRequest request = bulkRequest.requests.get(i);
-            if (request instanceof DocumentRequest) {
-                DocumentRequest req = (DocumentRequest) request;
-
-                if (addFailureIfIndexIsUnavailable(req, bulkRequest, responses, i, concreteIndices, metaData)) {
-                    continue;
+            if (request == null) {
+                continue;
```

Comment:
it does happen, I only made it explicit and scary, before we were relying on the instanceof check to return false, which happened only when we set the request to false. I can add a comment, sure!